test/python/algorithms/evolvers/test_pvqd.py

```diff
@@ -0,0 +1,225 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for PVQD."""
+
+from functools import partial
+from ddt import ddt, data, unpack
+import numpy as np
+
+from qiskit.test import QiskitTestCase
+
+from qiskit import BasicAer
+from qiskit.circuit import QuantumCircuit, Parameter, Gate
+from qiskit.algorithms.evolvers import EvolutionProblem
+from qiskit.algorithms.evolvers.pvqd import PVQD
+from qiskit.algorithms.optimizers import L_BFGS_B, GradientDescent, SPSA, OptimizerResult
+from qiskit.circuit.library import EfficientSU2
+from qiskit.opflow import X, Z, I, MatrixExpectation
+
+
+# pylint: disable=unused-argument, invalid-name
+def gradient_supplied(fun, x0, jac, info):
+    """A mock optimizer that checks whether the gradient is supported or not."""
+    result = OptimizerResult()
+    result.x = x0
+    result.fun = 0
+    info["has_gradient"] = jac is not None
+
+    return result
+
+
+class WhatAmI(Gate):
+    """An custom opaque gate that can be inverted but not decomposed."""
+
+    def __init__(self, angle):
+        super().__init__(name="whatami", num_qubits=2, params=[angle])
+
+    def inverse(self):
+        return WhatAmI(-self.params[0])
+
+
+@ddt
+class TestPVQD(QiskitTestCase):
+    """Tests for the pVQD algorithm."""
+
+    def setUp(self):
+        super().setUp()
+        self.sv_backend = BasicAer.get_backend("statevector_simulator")
+        self.qasm_backend = BasicAer.get_backend("qasm_simulator")
+        self.expectation = MatrixExpectation()
+        self.hamiltonian = 0.1 * (Z ^ Z) + (I ^ X) + (X ^ I)
+        self.observable = Z ^ Z
```

Comment:
Shall we also test with other operator types? E.g. `PauliOp`.

Reply:
I'm not sure we have to, since this just uses the `PauliEvolutionGate` which itself is tested with different ops 🙂 but I can add one 👍🏻